src/main/java/com/endava/internship/webapp/controller/DepartmentController.java

```diff
@@ -0,0 +1,42 @@
+package com.endava.internship.webapp.controller;
+
+import com.endava.internship.webapp.model.Department;
+import com.endava.internship.webapp.service.DepartmentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/departments")
+public class DepartmentController {
+
+    @Autowired
+    private DepartmentService departmentService;
+
+    @GetMapping
+    public List<Department> getAllDepartments() {
+        return departmentService.getAllDepartments();
+    }
+
+    @GetMapping("/{department_id}")
+    public Department getDepartmentById(@PathVariable("department_id") int id) {
+        return departmentService.getDepartmentById(id);
+    }
+
+    @PostMapping
```

Comment:
```suggestion
    @PostMapping
        public Department saveDepartment(@RequestBody **final** Department department) {
        return departmentService.saveDepartment(department);
    }
```
please add it everywhere

Reply:
Done